src/main/recorder.ts

```diff
@@ -209,10 +212,9 @@ const glob = require('glob');
             // It's a bit hacky that we async wait for 2 seconds for OBS to 
             // finish up with the video file. Maybe this can be done better. 
             setTimeout(async () => {
-                const bufferedVideo = getNewestVideo(this._bufferStorageDir);
+                const bufferedVideo = await getNewestVideo(this._bufferStorageDir);
                 await cutVideo(bufferedVideo, this._storageDir, metadata.duration);
-                writeMetadataFile(this._storageDir, metadata);  
-                resolve();       
```

Comment:
We've lost the call to resolve here. Don't we need that? writeMetadataFile now returns a resolved promise, is that enough to resolve the promise on line 207? I'm new to this stuff - so keen to understand!

Reply:
`.then()` takes a callback that will be called when the promise resolves, and in this case that would be the outer promise that will be resolved by that.

Basically, it has the same effect as doing:

```typescript
writeMetadataFile(this._storageDir, metadata).then(() => {
    resolve()
});
```

But without the extra callback (which just calls .. a .. callback).